cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -34,3 +39,148 @@ def test_cluster_single_db(cluster_with_single_db, logger, attributes, cfy):
                      cert_path=mgr2.local_ca)
 
     check_managers(mgr1, mgr2)
+
+
+def test_queue_node_failover(cluster_with_single_db, logger,
+                             module_tmpdir, attributes, ssh_key, cfy):
+    hello_world = prepare_cluster_with_agent(
+        cluster_with_single_db, logger,
+        module_tmpdir, attributes, ssh_key, cfy
+    )
+    broker1, broker2, broker3, db, mgr1, mgr2 = cluster_with_single_db
+    validate_cluster_and_agents(mgr1)
+    agent_broker_ip = verify_agent_broker_connection_and_get_broker_ip(mgr1)
+
+    # stop the rabbitmq service in the agent's broker node
+    agent_broker = None
+    for broker in [broker1, broker2, broker3]:
+        if broker.private_ip_address == agent_broker_ip:
+            agent_broker = broker
+            break
+    with agent_broker.ssh() as broker_ssh:
+        broker_ssh.run('sudo service cloudify-rabbitmq stop')
+
+    # the agent should now pick another broker
+    validate_cluster_and_agents(mgr1, expected_broker_status='Degraded')
+    new_agent_broker_ip = \
+        verify_agent_broker_connection_and_get_broker_ip(mgr1)
+    assert new_agent_broker_ip != agent_broker_ip
+
+    new_agent_broker = None
+    for broker in [broker1, broker2, broker3]:
+        if broker.private_ip_address == new_agent_broker_ip:
+            new_agent_broker = broker
+            break
+
+    # spin up 3 deployments
+    for i in range(3):
+        deployment_id = '{0}-{1}'.format(hello_world.deployment_id, i)
+        mgr1.client.deployments.create(
+            deployment_id=deployment_id,
+            blueprint_id=hello_world.blueprint_id,
+            inputs=hello_world.inputs)
+        mgr1.run_command(
+            'cfy executions start install -d {deployment_id} -t {tenant} '
+            '--timeout 900'.format(deployment_id=deployment_id,
+                                   tenant=hello_world.tenant))
+
+    # bring the stopped rabbit back up
+    with agent_broker.ssh() as broker_ssh:
+        broker_ssh.run('sudo service cloudify-rabbitmq start')
+
+    # verify that the resurrected rabbit shares the load
+    with new_agent_broker.ssh() as broker_ssh:
+        for broker in [broker1, broker2, broker3]:
+            num_queues = broker_ssh.run(
+                'sudo -u rabbitmq rabbitmqctl -n rabbit@{0} list_queues'
```

Comment:
Doesn't this list all queues anyway? Is there a way to find out what the primary node for a queue is?